### PR TITLE
Tensor API changes

### PR DESCRIPTION
**Pull request by fiona-gladwin**
- Introduce rocal_api_tensor.h file which has rocalTensor and rocalTensorList class to be exposed to the user
- Rename internal rocalTensor instances to Tensor

### CODE REVIEW

File: rocAL/rocAL/include/api/rocal_api_types.h [hunk 1]
```diff
@@ -121,7 +118,8 @@ enum RocalTensorLayout
     ROCAL_NHWC = 0,
     ROCAL_NCHW = 1,
     ROCAL_NFHWC = 2,
-    ROCAL_NFCHW = 3
+    ROCAL_NFCHW = 3,
+    ROCAL_NONE = 4
```
Comment (LakshmiKumar23) [inline rocAL/rocAL/include/api/rocal_api_types.h]: @fiona-gladwin what is the use case for this?

File: rocAL/rocAL/source/api/rocal_api_data_transfer.cpp [hunk 2]
```diff
@@ -122,7 +122,7 @@ ROCAL_API_CALL rocalSetOutputs(RocalContext p_context, unsigned int num_of_outpu
     auto context = static_cast<Context *>(p_context);
     std::vector<Image*> output_images_vector ;
     for (auto& it : output_images) {
-        auto img = static_cast<Image*>(it);
+        auto img = static_cast<Tensor *>(it);
```
Comment (LakshmiKumar23) [inline rocAL/rocAL/source/api/rocal_api_data_transfer.cpp:125]: @fiona-gladwin Is this supposed to ```Tensor*```

File: rocAL/rocAL/source/pipeline/tensor.cpp [hunk 3]
```diff
@@ -317,25 +317,45 @@ unsigned rocalTensor::copy_data(hipStream_t stream, void *host_memory, bool sync
...
-        // copy from device to host
-        hipError_t status;
-        if ((status = hipMemcpyDtoH((void *)user_buffer, _mem_handle, _info.data_size())))
-            THROW("copy_data::hipMemcpyDtoH failed: " + TOSTR(status))
-    } else
+        if (_info._mem_type == RocalMemType::HIP) {
+            // copy from device to device
+            hipError_t status;
+            if ((status = hipMemcpyDtoD((void *)user_buffer, _mem_handle, _info.data_size())))
+                THROW("copy_data::hipMemcpyDtoH failed: " + TOSTR(status))
```
Comment (LakshmiKumar23) [inline rocAL/rocAL/source/pipeline/tensor.cpp]: @fiona-gladwin Error thrown is wrong. Change to ```hipMemcpyDtoD```

File: rocAL/rocAL/source/api/rocal_api_data_transfer.cpp (hunk 2 above)
Comment (fiona-gladwin) [reply rocAL/rocAL/source/api/rocal_api_data_transfer.cpp:125]: Yes all Image references will be changed to Tensor

File: rocAL/rocAL/source/pipeline/tensor.cpp (hunk 3 above)
Comment (fiona-gladwin) [reply rocAL/rocAL/source/pipeline/tensor.cpp]: Done

File: rocAL/rocAL/include/api/rocal_api_types.h (hunk 1 above)
Comment (fiona-gladwin) [reply rocAL/rocAL/include/api/rocal_api_types.h]: This layout will be used for tensors that do not have a specific layout (other than image / video)

File: rocAL/rocAL/include/api/rocal_api_tensor.h [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#ifndef MIVISIONX_ROCAL_API_TENSOR_H
+#define MIVISIONX_ROCAL_API_TENSOR_H
+#include "rocal_api_types.h"
+
+class rocalTensor {
```
Comment (rrawther) [inline rocAL/rocAL/include/api/rocal_api_tensor.h:27]: probably we should rename it as rocalTensorApi

File: rocAL/rocAL/include/api/rocal_api_tensor.h [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    virtual RocalTensorLayout layout() = 0;
+    virtual RocalTensorBackend backend() = 0;
+    virtual RocalTensorOutputType data_type() = 0;
+    virtual size_t data_size() = 0;
+    virtual RocalROICordsType roi_type() = 0;
+    virtual RocalROICords *get_roi() = 0;
+    virtual std::vector<size_t> shape() = 0;
+};
+
+class rocalTensorList {
```
Comment (rrawther) [inline rocAL/rocAL/include/api/rocal_api_tensor.h:44]: rename to rocalTensorListApi

File: rocAL/rocAL/include/augmentations/color_augmentations/node_brightness.h [hunk 6]
```diff
@@ -29,7 +29,7 @@ THE SOFTWARE.
 class BrightnessNode : public Node
 {
 public:
-    BrightnessNode(const std::vector<rocalTensor *> &inputs, const std::vector<rocalTensor *> &outputs);
+    BrightnessNode(const std::vector<Tensor *> &inputs, const std::vector<Tensor *> &outputs);
```
Comment (rrawther) [inline rocAL/rocAL/include/augmentations/color_augmentations/node_brightness.h:32]: why is this renamed? Because of that a lot of files has been changed. Instead we can just rename the newly introduced api structures. Please revert back

Comment (fiona-gladwin) [reply rocAL/rocAL/include/augmentations/color_augmentations/node_brightness.h:32]: In general we do not have the prefix rocal to any of the internal functions or the class/struct. Since we wanted to expose the class externally we had named it as rocalTensor and rocalTensorList initially. With this PR we will change all instances internally from Image->Tensor and ImageInfo->TensorInfo and the external API exposed to the user will be rocalTensor and rocalTensorList.
This will be just a one time change and we can follow this convention for upcoming PR.

**Review by rrawther — APPROVED** [review]

File: rocAL/rocAL/include/api/rocal_api_types.h (hunk 1 above)
Comment (rrawther) [reply rocAL/rocAL/include/api/rocal_api_types.h]: Please remove ROCAL_NONE if it is not used in the code. This is confusing from the layout perspective
